### PR TITLE
Add smarter host and port discovery for SocketUtil

### PR DESCRIPTION
**Pull request by drcrallen**

### CODE REVIEW

File: server/src/main/java/io/druid/server/DruidNode.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ private void init(String serviceName, String host, Integer port)
 
     if(host == null && port == null) {
       host = getDefaultHost();
-      port = -1;
+      port = SocketUtil.findOpenPort(8080, host);
```
Comment (xvrl) [inline server/src/main/java/io/druid/server/DruidNode.java]: why are we changing this one?

Comment (drcrallen) [inline server/src/main/java/io/druid/server/DruidNode.java]: If host and port are both null unit test will fail because port = -1 is not a valid setting.

**Comment by drcrallen** [comment]
hmmm... other unit tests failing, closing until resolved

**Comment by drcrallen** [comment]
Yeah, the unit tests are not at all friendly to this change. Leaving closed since it seems to change prior behavior too much.

Opening issue instead.

**Pull request closed without merging**